src/CorePolicyRamCloud.cc

```diff
@@ -0,0 +1,94 @@
+/* Copyright (c) 2009-2016 Stanford University
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR(S) DISCLAIM ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL AUTHORS BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+#include "ShortMacros.h"
+#include "Cycles.h"
+#include "Arachne/Arachne.h"
+#include "Arachne/CorePolicy.h"
+#include "CorePolicyRamCloud.h"
+#include "FileLogger.h"
+
+using namespace RAMCloud;
+
+/* Is the core load estimator running? */
+bool ramCloudLoadEstimatorRunning = false;
+
+/*
+ * Update threadClassCoreMap when a new core is added.  Takes in the coreId
+ * of the new core.  Assigns the first core to the dispatch class, the
+ * hypertwin of that one to the dispatchHyperTwin class, and all others to the
+ * default class.
+ *
+ * \param coreId
+ *     The coreId of the new core.
+ */
+void CorePolicyRamCloud::addCore(int coreId) {
+    CoreList* dispatchEntry = threadClassCoreMap[dispatchClass];
+    // Assign the dispatch thread to the first core that comes up.
+    if (dispatchEntry->numFilled == 0) {
+        dispatchEntry->map[0] = coreId;
+        dispatchEntry->numFilled++;
+        dispatchHyperTwin = getHyperTwin(sched_getcpu());
```

Comment:
Huh? What makes you think sched_getcpu will return the desired core id? Let's discuss.